onnxruntime/core/providers/migraphx/migraphx_execution_provider.h

```diff
@@ -100,6 +101,7 @@ class MIGraphXExecutionProvider : public IExecutionProvider {
   migraphx::target t_;
   OrtMutex mgx_mu_;
   hipStream_t stream_ = nullptr;
+  mutable char model_path_[4096] = {};  // Reserved for max path length
```

Comment:
Please change it to `std::filesystem::path` or `std::string`. Your assumption about the max path needs to be corrected -NTFS can have paths up to 64K characters.